playground/backend/internal/code_processing/code_processing.go

```diff
@@ -55,32 +57,42 @@ func Process(ctx context.Context, cacheService cache.Cache, lc *fs_tool.LifeCycl
 	go cancelCheck(ctx, pipelineId, cancelChan, cacheService)
 
 	// build executor for validate and compile steps
-	exec := compileBuilder.Build()
+	executor := compileBuilder.Build()
 
 	// validate
 	logger.Infof("%s: Validate() ...\n", pipelineId)
-	validateFunc := exec.Validate()
+	validateFunc := executor.Validate()
 	go validateFunc(successChannel, errorChannel)
 
-	if err := processStep(ctxWithTimeout, pipelineId, cacheService, cancelChan, successChannel, nil, errorChannel, pb.Status_STATUS_VALIDATION_ERROR, pb.Status_STATUS_PREPARING); err != nil {
+	if err := processStep(ctxWithTimeout, pipelineId, cacheService, cancelChan, successChannel, nil, nil, errorChannel, pb.Status_STATUS_VALIDATION_ERROR, pb.Status_STATUS_PREPARING); err != nil {
 		return
 	}
 
 	// prepare
 	logger.Infof("%s: Prepare() ...\n", pipelineId)
-	prepareFunc := exec.Prepare()
+	prepareFunc := executor.Prepare()
 	go prepareFunc(successChannel, errorChannel)
 
-	if err := processStep(ctxWithTimeout, pipelineId, cacheService, cancelChan, successChannel, nil, errorChannel, pb.Status_STATUS_PREPARATION_ERROR, pb.Status_STATUS_COMPILING); err != nil {
+	if err := processStep(ctxWithTimeout, pipelineId, cacheService, cancelChan, successChannel, nil, nil, errorChannel, pb.Status_STATUS_PREPARATION_ERROR, pb.Status_STATUS_COMPILING); err != nil {
 		return
 	}
 
 	// compile
 	logger.Infof("%s: Compile() ...\n", pipelineId)
-	compileCmd := exec.Compile(ctxWithTimeout)
-	go processCmd(compileCmd, successChannel, errorChannel, dataChannel)
+	compileCmd := executor.Compile(ctxWithTimeout)
+	go func(cmd *exec.Cmd, successCh chan bool, errCh chan error, dataCh chan interface{}) {
+		// TODO separate stderr from stdout
```

Comment:
It is not the best idea to leave to-do's, most likely some reviewer ask to to do it

Reply:
Let's leave a Jira ticket number here with a corresponding task